### PR TITLE
Convert question score partials to components

### PR DESCRIPTION
**Pull request by jonatanschroeder**

### CODE REVIEW

File: apps/prairielearn/src/components/QuestionScore.html.ts [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+  const hasAutoAndManualPoints =
+    assessment_question.max_auto_points &&
+    (assessment_question.max_manual_points ||
+      instance_question.manual_points ||
+      instance_question.requires_manual_grading);
+
+  return html`
+    <div class="card mb-4" id="question-score-panel">
+      <div class="card-header bg-secondary text-white">
+        Question ${instance_question_info.question_number}
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/components/QuestionScore.html.ts:57]: The original EJS had a bug here that seems to have been going on for a long time. This was using `instance_question.question_number`, which is undefined. As a result, this component did not have the question number listed in the code. I have added this back, assuming this is indeed the expected functionality.

File: apps/prairielearn/src/components/QuestionScore.html.ts [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+  component,
+}: {
+  instance_question: Pick<
+    InstanceQuestion,
+    'auto_points' | 'manual_points' | 'points' | 'status' | 'requires_manual_grading'
+  >;
+  assessment_question: Pick<
+    AssessmentQuestion,
+    'max_auto_points' | 'max_manual_points' | 'max_points'
+  >;
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/components/QuestionScore.html.ts:354]: There are some pages using this component that override some of the types of individual variables. The `Pick` options here are to ensure that at least the columns we need are still properly defined. Also, some pages for now combine values like `max_*_points` into the instance question object, so this simplifies the handling of these pages until they are transitioned to a stronger type.

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts [hunk 3]
```diff
@@ -180,15 +181,14 @@ export function InstructorAssessmentInstance({
...
-                      >/<span id="total-max-points" class="text-muted"
-                        >${resLocals.assessment_instance.max_points.toString()}</span
-                      ></small
-                    >
+                    <span id="total-points">
+                      ${formatPoints(resLocals.assessment_instance.points)}
+                    </span>
+                    <small>
+                      /<span id="total-max-points" class="text-muted"
+                        >${formatPoints(resLocals.assessment_instance.max_points)}
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts:189]: Presumably this is mushed onto one line to keep from introducing spaces where there shouldn't be any, is that right?

File: apps/prairielearn/src/pages/partials/pointsFormatter.ejs
Comment (nwalters512) [inline apps/prairielearn/src/pages/partials/pointsFormatter.ejs:1]: 🎉 happy to see this deleted!

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts [hunk 4]
```diff
@@ -345,11 +345,11 @@ export function InstructorAssessmentInstance({
...
                       </td>
                       <td class="text-center">
-                        ${renderEjs(
-                          import.meta.url,
-                          "<%- include('../partials/instanceQuestionPoints') %>",
-                          { instance_question, component: 'auto' },
-                        )}
+                        ${InstanceQuestionPoints({
+                          instance_question,
+                          assessment_question: instance_question, // Required fields are present in instance_question
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts:350]: This is strange to me, but I don't have a better suggestion.

**Review by nwalters512 — APPROVED** [review]
As usual with these, I didn't do a careful comparison between old partials and new code. If you'd like me to focus on anything specific, please let me know.

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts (hunk 3 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts:189]: Indeed. The score should stay with the slash that precedes it.

**Pull request merged**